### PR TITLE
WIP: Owls 68520

### PR DESCRIPTION
**Pull request by vkraemer**
these changes address https://jira.oraclecorp.com/jira/browse/OWLS-68520

This includes changes to the integration tests to validate the change.

wercker runs to green: https://app.wercker.com/Oracle/weblogic-kubernetes-operator/runs/integration-test-java/5c3cf9870f540f00258669ed

### CODE REVIEW

**Comment by rjeberhard** [comment]
Everything here looks good.  To complete the work, you should add unit-tests (PodHelperTest, ServiceHelperTest) that validate that we generate pods and services with appropriately decorated names when the domain has servers with ".".  Finally, reach out to @rosemarymarano.  We should add the rules of server name to K8s resource name conversion to our public doc.

File: integration-tests/src/test/java/oracle/kubernetes/operator/utils/Domain.java [hunk 1]
```diff
@@ -782,14 +783,15 @@ private void createLoadBalancer() throws Exception {
 
   private void callShellScriptByExecToPod(String username, String password, String webappName)
       throws Exception {
+    String asn = adminServerName.toLowerCase(Locale.ENGLISH).replace('_', '-').replace('.', '-');
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/kubernetes/operator/utils/Domain.java]: I think you can replace the characters '_ 'and '.' with '-' for adminServerName and managedServerNameBase  in the map itself in initialize() method after loading the yaml to map.

File: integration-tests/src/test/java/oracle/kubernetes/operator/utils/Domain.java [hunk 2]
```diff
@@ -855,7 +857,9 @@ private void callWebAppAndCheckForServerNameInResponse(
     // map with server names and boolean values
     HashMap<String, Boolean> managedServers = new HashMap<String, Boolean>();
     for (int i = 1; i <= TestUtils.getClusterReplicas(domainUid, clusterName, domainNS); i++) {
-      managedServers.put(domainUid + "-" + managedServerNameBase + i, new Boolean(false));
+      String msnb =
+          managedServerNameBase.toLowerCase(Locale.ENGLISH).replace('_', '-').replace('.', '-');
+      managedServers.put(domainUid + "-" + msnb + i, new Boolean(false));
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/kubernetes/operator/utils/Domain.java]: I think you can replace the characters '_ 'and '.' with '-' for adminServerName and managedServerNameBase  in the map itself in initialize() method after loading the yaml to map.

**Comment by markxnelson** [comment]
@LorenKonkus  @rjeberhard I thought we had discussed this and agreed that we should not support periods in the names of servers.

**Comment by vkraemer** [comment]
@markxnelson ( @rjeberhard @LorenKonkus )

Are you implying that the currently shipping behavior is the expected behavior... that would lead me to think that this whole change is moot OR that you expect a different behavior wrt how that error is presented to the user.

Please clarify your expectation

**Comment by rjeberhard** [comment]
We've had a lot of conversations...  I'm vaguely remembering this one, but not the decision.

@moreaut @tbarnes-us Did one of you mention that server names with periods was legal but caused problems? If so, why did we never limit the legal names in WebLogic?

**Pull request closed without merging**